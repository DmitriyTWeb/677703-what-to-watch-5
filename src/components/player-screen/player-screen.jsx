import React from "react";
import PropTypes from "prop-types";
import {filmType} from "../../custom-prop-types";
import {connect} from "react-redux";

import FullVideoPlayer from "../full-video-player/full-video-player.jsx";
import PlayButton from "../play-button/play-buttton";
import TimeElapsed from "../time-elapsed/time-elapsed";

import withVideoHandling from "../../hocs/with-video-handling/with-video-handling";
import withTimeElapsedHandling from "../../hocs/with-time-elapsed-handling/with-time-elapsed-handling";
import {getAllFilms} from "../../store/selectors/selectors";

const FullVideoPlayerWrapped = withVideoHandling(FullVideoPlayer);
const TimeElapsedWrapped = withTimeElapsedHandling(TimeElapsed);


const PlayerScreen = (props) => {
  const {
    allFilms,
    match,
    onVideoMount,
    onPlayButtonClick,
    onFullscreenButtonClick,
    onExitButtonClick,
    isPlaying,
    videoRef,
  } = props;

  const currentId = +match.params.id;
  const currentFilm = allFilms.find((film) => film.id === currentId);
  const {
    fullImg,
    video,
  } = currentFilm;

  return (
    <div className="player">
      <FullVideoPlayerWrapped
        src={video}
        fullImg={`${fullImg}`}
        onVideoMount={onVideoMount}
      />

      <button type="button" className="player__exit" onClick={onExitButtonClick}>
        Exit
      </button>

      <div className="player__controls">
        <div className="player__controls-row">
          <div className="player__time">
            <progress className="player__progress" value="30" max="100"></progress>
            <div className="player__toggler" style={{left: `30%`}}>Toggler</div>
          </div>
          <TimeElapsedWrapped
            videoRef={videoRef}
          />
        </div>

        <div className="player__controls-row">
          <PlayButton
            isPlaying={isPlaying}
            onPlayButtonClick={onPlayButtonClick}
          />

          <div className="player__name">
            {currentFilm.title}
          </div>

          <button type="button" className="player__full-screen"
            onClick={onFullscreenButtonClick}>
            <svg viewBox="0 0 27 27" width="27" height="27">
              <use xlinkHref="#full-screen"></use>
            </svg>
            <span>Full screen</span>
          </button>
        </div>
      </div >
    </div >
  );
};

PlayerScreen.propTypes = {
  allFilms: PropTypes.arrayOf(filmType).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }).isRequired,
  }).isRequired,
  onVideoMount: PropTypes.func.isRequired,
  onPlayButtonClick: PropTypes.func.isRequired,
  onFullscreenButtonClick: PropTypes.func.isRequired,
  onExitButtonClick: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool.isRequired,
  videoRef: PropTypes.object,
};


const mapStateToProps = (state) => ({
  allFilms: getAllFilms(state),
});

export {PlayerScreen};
export default connect(mapStateToProps)(PlayerScreen);
