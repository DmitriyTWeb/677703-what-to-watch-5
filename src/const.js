export const RatingDesc = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const DEFAULT_RENDERED_FILMS_COUNT = 8;
export const PLAYING_TIMEOUT = 1000;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  MYLIST: `/mylist`,
  FILMS_ID: `/films/:id`,
  PLAYER_ID: `/player/:id`,
  REVIEW_ID: `/films/:id/review`,
  ERROR: `/error`,
};

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  PROMO: `/films/promo`,
};
