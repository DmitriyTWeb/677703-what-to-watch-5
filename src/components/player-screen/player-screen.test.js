import React from "react";
import renderer from "react-test-renderer";
import {PlayerScreen} from "./player-screen";
import {mockFilm, mockMatchParamsId} from "../../const.js";

const mockFilms = [mockFilm, mockFilm, mockFilm];
const noop = () => { };

jest.mock(`../full-video-player/full-video-player.jsx`, () => () => `FullVideoPlayer`);
jest.mock(`../play-button/play-buttton`, () => () => `PlayButton`);
jest.mock(`../time-elapsed/time-elapsed`, () => () => `TimeElapsed`);
jest.mock(`../../hocs/with-video-handling/with-video-handling`, () => () => `withVideoHandling`);
jest.mock(`../../hocs/with-time-elapsed-handling/with-time-elapsed-handling`, () => () => `withTimeElapsedHandling`);


it(`Should PlayerScreen render correctly`, () => {
  const tree = renderer.create(
      <PlayerScreen
        allFilms={mockFilms}
        match={mockMatchParamsId}
        onVideoMount={noop}
        onPlayButtonClick={noop}
        onFullscreenButtonClick={noop}
        onExitButtonClick={noop}
        isPlaying={false}
        videoRef={{}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
