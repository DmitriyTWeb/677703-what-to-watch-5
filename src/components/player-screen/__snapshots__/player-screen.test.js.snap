// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PlayerScreen render correctly 1`] = `
<div
  className="player"
>
  <withVideoHandling
    fullImg="img/the-grand-budapest-hotel-bg.jpg"
    onVideoMount={[Function]}
    src="https://some-link"
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value="30"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "30%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <withTimeElapsedHandling
        isPlaying={false}
        videoRef={Object {}}
      />
    </div>
    <div
      className="player__controls-row"
    >
      PlayButton
      <div
        className="player__name"
      >
        The Grand Budapest Hotel
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
